#! /usr/bin/env python
# Forearm camera configuration

PACKAGE='canon_vbc50i'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#       Name                    Type      Reconfiguration level             Description                                  Default    Min   Max
gen.add("hostname",    str_t,    SensorLevels.RECONFIGURE_CLOSE,   "The hostname where the camera can be found",               "192.168.0.1")
gen.add("subsampling", int_t,    SensorLevels.RECONFIGURE_RUNNING, "Image size as the binay power of the fraction of full size (1:1, 1:2, 1:4 or 1:8)", 0,      0,     3)
gen.add("frame_id",    str_t,    SensorLevels.RECONFIGURE_RUNNING, "The frame in which laser scans will be returned",          "camera")
gen.add("pause",       bool_t,   SensorLevels.RECONFIGURE_RUNNING, "Pause/resume the video streaming",                         False)
gen.add("night_mode",  bool_t,   SensorLevels.RECONFIGURE_RUNNING, "Turn on/off the night mode",                               False)
gen.add("infrared",    bool_t,   SensorLevels.RECONFIGURE_RUNNING, "Turn on/off the infrared light",                           False)
gen.add("autoexp",     bool_t,   SensorLevels.RECONFIGURE_RUNNING, "Turn on/off the autoexposure control",                     True)
gen.add("digital_zoom",bool_t,   SensorLevels.RECONFIGURE_RUNNING, "Turn on/off digital zoom",                     False)
gen.add("focus_mode",  int_t,    SensorLevels.RECONFIGURE_RUNNING, "Set focus mode, 0 for auto",                               0,       0,      4)
gen.add("record_dir",  str_t,    SensorLevels.RECONFIGURE_RUNNING, "Directory where the video frame are recorded",             "/tmp")
gen.add("record",      bool_t,   SensorLevels.RECONFIGURE_RUNNING, "Turn on/off the video recording",                          False)
gen.add("pan_ang",     double_t, SensorLevels.RECONFIGURE_RUNNING, "Current pan angle (deg).",                                 0,       -170,   170)
gen.add("tilt_ang",    double_t, SensorLevels.RECONFIGURE_RUNNING, "Current tilt angle (deg).",                                0,       -90,    10)
gen.add("zoom_ang",    double_t, SensorLevels.RECONFIGURE_RUNNING, "Current zoom, expressed as a field of view angle (deg).",  41.26,   1.97,   41.26)
gen.add("fps",         int_t,    SensorLevels.RECONFIGURE_RUNNING, "Requested frame per second",                               10,      1,      25)
gen.add("aperture",    int_t,    SensorLevels.RECONFIGURE_RUNNING, "Aperture value.",                                          7,       1,      15)
gen.add("inv_shutter", int_t,    SensorLevels.RECONFIGURE_RUNNING, "Shutter speed, expressed as a fraction of second.",        128,     1,      250)
gen.add("gain",        int_t,    SensorLevels.RECONFIGURE_RUNNING, "Gain.",                                                    128,     1,      256)
gen.add("pan_speed",   int_t,    SensorLevels.RECONFIGURE_RUNNING, "Panning speed in deg/s",                                   10,      1,      1024)
gen.add("tilt_speed",   int_t,    SensorLevels.RECONFIGURE_RUNNING, "Tilting speed in deg/s",                                  10,      1,      1024)
gen.add("zoom_speed",   int_t,    SensorLevels.RECONFIGURE_RUNNING, "Zooming speed in deg/s",                                  10,      1,      1024)


exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "CanonParams"))

